adoc/extensions/sycl_khr_queue_empty_query.adoc

```diff
@@ -0,0 +1,78 @@
+[[sec:khr-queue-empty-query]]
+= sycl_khr_queue_empty_query
+
+This extension allows developers to query the queue's emptiness, meaning if all
+commands submitted to a queue have been completed.
+
+[[sec:khr-queue-empty-query-dependencies]]
+== Dependencies
+
+This extension has no dependencies on other extensions.
+
+[[sec:khr-queue-empty-query-feature-test]]
+== Feature test macro
+
+An implementation supporting this extension must predefine the macro
+[code]#SYCL_KHR_QUEUE_EMPTY_QUERY# to one of the values defined in the table
+below.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|Initial version of this extension.
+|===
+
+
+[[sec:khr-queue-empty-query-funct]]
+== New Queue Function to Query Emptiness
+
+This extension adds the following function to the [code]#sycl::queue# class,
+which provides information about the emptiness of the queue.
+
+'''
+
+.[apidef]#queue::khr_empty#
+[source,role=synopsis,id=api:queue-khr-empty]
+----
+bool khr_empty() const
+----
+
+_Returns:_ [code]#true# if all <<command,commands>> enqueued on this queue have
+completed, [code]#false# otherwise.
+
+{note} Since the implementation executes commands asynchronously, the returned
+value is a snapshot in time.
+{endnote}
+
+'''
+
+[[sec:khr-queue-empty-query-example]]
+== Example
+
+The example below demonstrates the usage of this extension.
+
+[source,,linenums]
+----
+#include <iostream>
+#include <algorithm>
+#include <sycl/sycl.hpp>
+int main() {
+  // Pool of queues, one per device
+  std::vector<sycl::queue> Qs;
+  for (sycl::device &d : sycl::platform{}.get_devices())
+    Qs.push_back(sycl::queue(d));
+
+  // Useful recipe for load-balancing
+  auto it = std::find_if(Qs.begin(), Qs.end(),
+                         [](const sycl::queue &q) { return q.khr_empty(); });
```

Comment:
```suggestion
                         [](const sycl::queue& q) { return q.khr_empty(); });
```

Reply:
Did we commit to a clang-format at some point? I don't remember...

Reply:
Ok, we have one; I just didn't used it my mistake. I should try to find the script I did to format the `[API,code]` and commit it to the repo / in the CI.
I guess I tried long time ago `https://gitlab.khronos.org/sycl/Specification/-/merge_requests/730#note_387779` and then forgot about it...

Reply:
Ok, I am confirming that `https://github.com/KhronosGroup/SYCL-Docs/pull/292#issuecomment-1270531337` did fix the wrong syntax. (and fixed the include order)
Will try to put that as a PR / linter; maybe be helpful for other people.